search/includes/classes/class-search.php

```diff
@@ -885,13 +891,41 @@ public function get_statsd_request_mode_for_request( $url, $args ) {
 
 		// Bulk indexing
 		if ( '_bulk' === end( $path ) ) {
-			return 'index';
+			return $this->is_full_index( $path, $args ) ? 'full_index' : 'index';
 		}
 
 		// Unknown
 		return 'other';
 	}
 
+	/**
+	 * From the path will parse what type of indaxable is it. From the body of the request how many documents we are indexing.
+	 *
+	 * If the number of documents matches the number of corresponding objects in DB, will return true, false otherwise.
+	 */
+	private function is_full_index( array $path, array $args ) {
```

Comment:
The logic rules for this function seem very fragile, and it could end up being fairly expensive to run because it has to do a DB lookup for every call.

If we're trying to detect the duration of a full bulk index, for every post, we should put our stat tracking into the CLI command. Without having the start time, we'd only ever have the duration of the last request anyway (unless I'm totally misunderstanding the purpose of this code).

Reply:
Yeah that is difficult. My first implementation was in the command, but there we can't differentiate between full index and/or errors (because we don't get the EP command result within our vip command) - so @brandon-m-skinner  helped me by pointing this section out. 

But Yeah, I am not sure how those bulk requests are created and if they can be split into batches on bigger sets of documents, it would be a problem as well.